owasp-top10-2017-apps/a2/saidajaula-monster/app/app.py

```diff
@@ -17,39 +19,52 @@
                     os.environ.get('A2_DATABASE_PASSWORD'),
                     os.environ.get('A2_DATABASE_NAME'))
 
+SECRET_KEY = os.environ.get('SECRET_KEY')
 
 def login_admin_required(f):
     @wraps(f)
     def decorated_function(*args, **kwargs):
-        cookie = request.cookies.get("sessionId", "")
-        cookie = base64.b64decode(cookie).decode("utf-8")
-        cookie_separado = cookie.split('.')
-        if(len(cookie_separado) != 2):
-            return "Invalid cookie!"
-        hash_cookie = hashlib.sha256(cookie_separado[0].encode('utf-8')).hexdigest()
-        if (hash_cookie != cookie_separado[1]):
-            return redirect("/login")
-        j = json.loads(cookie_separado[0])
-        if j.get("permissao") != 1:
-            return "You don't have permission to access this route. You are not an admin. \n"
-        return f(*args, **kwargs)
+        cookie = request.cookies.get("access_token", "")
+        try:
+            token = jwt.decode(cookie, SECRET_KEY, algorithms='HS256')
+            if token["permissao"] != 1:
+                return "You don't have permission to access this route. You are not an admin. \n"
+            return f(*args, **kwargs)
+        except:
+            return redirect("/refresh-token")
+
     return decorated_function
 
 
 def login_required(f):
     @wraps(f)
     def decorated_function(*args, **kwargs):
-        cookie = request.cookies.get("sessionId", "")
-        cookie = base64.b64decode(cookie).decode("utf-8")
-        cookie_separado = cookie.split('.')
-        if(len(cookie_separado) != 2):
-            return "Invalid cookie! \n"
-        hash_cookie = hashlib.sha256(cookie_separado[0].encode('utf-8')).hexdigest()
-        if (hash_cookie != cookie_separado[1]):
-            return redirect("/login")
-        return f(*args, **kwargs)
+        cookie = request.cookies.get("access_token", "")
+        try:
+            jwt.decode(cookie, SECRET_KEY, algorithms='HS256')
+            return f(*args, **kwargs)
+        except:
+            return redirect("/refresh-token")
```

Comment:
Very good! But I find the route name `/refresh-token` a little confusing. Because what you need to do is to request a new **access token**, using the **refresh token** that you already have.